tasks/add-fbc-contribution/add-fbc-contribution.yaml

```diff
@@ -0,0 +1,104 @@
+---
+apiVersion: tekton.dev/v1
+kind: Task
+metadata:
+  name: add-fbc-contribution
+  labels:
+    app.kubernetes.io/version: "1.0.0"
+  annotations:
+    tekton.dev/pipelines.minVersion: "0.12.1"
+    tekton.dev/tags: release
+spec:
+  description: >-
+    Task to create internalrequests to add contributions to index images
+  params:
+    - name: snapshotPath
+      description: Path to the JSON string of the mapped Snapshot spec in the data workspace
+      type: string
+      default: "snapshot_spec.json"
+    - name: dataPath
+      description: Path to the JSON string of the merged data to use in the data workspace
+      type: string
+      default: "data.json"
+    - name: binaryImage
+      type: string
+      description: binaryImage value updated by update-ocp-tag task
+    - name: fromIndex
+      type: string
+      description: fromIndex value updated by update-ocp-tag task
+    - name: requestTimeout
+      type: string
+      default: "180"
+      description: InternalRequest timeout
+  results:
+    - name: fbcOptIn
+      description: Result to set weather the contribution is FBC opt in
+    - name: requestResultsFile
+      description: Internal Request results file
+    - name: requestMessage
+      description: Internal Request message
+    - name: requestReason
+      description: Internal Request reason
+  workspaces:
+    - name: data
+      description: workspace to read and save files
+  steps:
+    - name: add-contribution
+      image: quay.io/hacbs-release/release-utils:a9603c0ae1982be5e1142d44765b1475adc7ad40
+      script: |
+        #!/usr/bin/env sh
+        #
+        set -e
+
+        SNAPSHOT_PATH=$(workspaces.data.path)/$(params.snapshotPath)
+        DATA_FILE="$(workspaces.data.path)/$(params.dataPath)"
+        if [ ! -f "${DATA_FILE}" ] ; then
+            echo "No valid data file was provided."
+            exit 1
+        fi
+
+        # Set the value for requestResultsFile
+        echo "$(workspaces.data.path)/ir-$(context.taskRun.uid)-result.txt" > $(results.requestResultsFile.path)
```

Comment:
It's probably better to use `echo -n` to avoid the new line.

Reply:
Also, how is this file actually used? I see that it's provided as input to some other tasks, but I didn't dig deeper. My concern is that it's the output of the `internal-request` script, so it contains not only a json snippet, but also more than that. So those consuming tasks would also need to parse it somehow, right? I guess this will also be fixed in the open jira we have for saving the scripts's output as a file?

Reply:
Actually, didn't you say you were removing this file? :) Here: https://github.com/redhat-appstudio/release-service-catalog/pull/270#discussion_r1383236730

Reply:
The actual answer is - due to the generic approach of the `create-internal-request`, to avoid creating infinite results with the IIB result string, which is big and also can ´t fit in a single result, the option was save it in a file and use it when required by the next tasks. As this task is more like changing A to B, I did not change this, but for the next ones I might be able to do.

I could refactor all tasks and the pipeline in a single PR, but it would be too many files to review.

Reply:
I see, ok.

Reply:
BTW, this still has `echo` instead of `echo -n`.

Reply:
fixed.

Reply:
ok, cool. but I noticed another thing - added comment.